src/odemis/driver/orsay.py

```diff
@@ -2769,3 +2781,270 @@ def terminate(self):
+        self.referenced._set_value({"x": referenced_value, "y": referenced_value}, force_write=True)
+
+    def connectApertureDict(self):
+        """
+        (re)Connects the aperture dict to the Orsay server by subscribing listeners to all the aperture data and
+        updating the values.
+        """
+        # Remove the references to the old connectors.
+        self._apertureConnectors = {i: [] for i in range(self._lastApertureNmbr)}
+
+        for aprtr_nmbr in range(self._lastApertureNmbr):
+            # Subscribe to and update the value Lifetime
+            param = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.Lifetime")
+            self._addApertureListener(aprtr_nmbr, "Lifetime", param)
+            
+            # Subscribe to and update the value Size
+            param = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.Size")
+            self._addApertureListener(aprtr_nmbr, "Size", param)
+
+            # Subscribe to and update the value PositionX
+            param = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.PositionX")
+            self._addApertureListener(aprtr_nmbr, "PositionX", param)
+
+            # Subscribe to and update the value PositionY
+            param = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.PositionY")
+            self._addApertureListener(aprtr_nmbr, "PositionY", param)
+
+    def _addApertureListener(self, aperture_nmbr, param_type, parameter):
+        """
+        Subscribes _apertureListener to an Orsay parameter with the right input to update to the apertureDict when called.
+        Also initializes the value by calling the listener once.
+
+        :param aperture_nmbr (int): number of the aperture in the dict from 0 --> 30
+        :param param_type (type func): type of variable to be saved. Function used to convert str from the Orsay parameter.
+        :param parameter (Orsay Parameter): A parameter of the Orsay server.
+        :return:
+        """
+        listener = partial(self._apertureListener, aperture_nmbr, param_type)
+        parameter.Subscribe(listener)
+        self._apertureConnectors[aperture_nmbr].append(listener)
+        listener(parameter, attr_name="Actual")  # Make a fake call to the listener to initialize the value.
+
+    def _apertureListener(self, aperture_nmbr, param_type, parameter, attr_name="Actual"):
+        if attr_name != "Actual":
+            return
+        var_type = self._apertureVarTypes[param_type]
+        if param_type.startswith("Position"):
+            self._apertureDict[aperture_nmbr]["Position"][param_type[-1].lower()] = var_type(parameter.Actual)
+        else:
+            self._apertureDict[aperture_nmbr][param_type] = var_type(parameter.Actual)
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        When the aperture block needs to be moved to a position, or an aperture must be selected.
```

Comment:
how would you select an aperture with this?

Reply:
I updated the comment (removed the part about selecting an aperture). The rest is explained at the docstring of the class itself:

    """
    Represents the Aperture carrier of the Orsay FIB. This class allows controls of the current position of the
    aperture carrier using the move functions like a stage. This allows positioning of a certain aperture w.r.t. the
    ion beam. The aperture carrier is a plate which contains multiple apertures.
    Using the selectedAperture VA an aperture can be selected using the number it is identified with. The size of the
    currently selected aperture can be found using the sizeSelectedAperture VA.

    Class contains the metadata MD_APERTURES_INFO which is a dict containing for each aperture the lifetime,
    size and position in x and y.
    """